### PR TITLE
allow reference unsupported cfn resources

### PR DESCRIPTION
**Pull request by pinzon**
## Motivation
Addresses #11362. LocalStack already allows the deployment of templates with unsupported resource types. (Check `CFN_IGNORE_UNSUPPORTED_RESOURCE_TYPES` in [docs](https://docs.localstack.cloud/references/configuration/)). With these changes any reference to these types returns a mock value.

## Changes
- Before raising DependencyNotYetSatisfied the template deployer check if the resource type is supported

## Testing
- New `only_localstack` test.

### CODE REVIEW

File: tests/aws/templates/cfn_ref_unsupported.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+Resources:
+  CodeDeployApplication:
+    Type: AWS::CodeDeploy::Application
```
Comment (dominikschubert) [inline tests/aws/templates/cfn_ref_unsupported.yml:3]: nit: I'd suggest using something that we're sure won't be implemented in the future here instead, e.g. `AWS::LocalStack::Unknown`

**Review by dominikschubert — COMMENTED** [review]
Can you maybe explain a bit the decision behind preferring to patch this in resolve_refs_recursively, rather than actually setting the physical resource id on the resource after the skipped deployment? 

I'd also like to see at least one more test here to show that other access patterns besides a Ref in the Outputs section works as well, e.g. wrapped in an intrinsic function in a resource properties section.

File: tests/aws/templates/cfn_ref_unsupported.yml (hunk 1 above)
Comment (pinzon) [reply tests/aws/templates/cfn_ref_unsupported.yml]: Thanks for the suggestion. I extended the test to use an intrinsic function an a development. Also moved the patch outside the recursive function.

File: localstack-core/localstack/services/cloudformation/engine/template_deployer.py [hunk 2]
```diff
@@ -1270,6 +1285,8 @@ def apply_change(self, change: ChangeConfig, stack: Stack) -> None:
                 resource_provider, resource, resource_provider_payload
             )
         else:
+            resource["PhysicalResourceId"] = MOCK_REFERENCE
+            resource["Ref"] = MOCK_REFERENCE
```
Comment (dominikschubert) [inline localstack-core/localstack/services/cloudformation/engine/template_deployer.py:1289]: Why do we need to set the Ref here? :thinking:  Is this somehow overriding the physical resource id lookup if set?

Comment (pinzon) [reply localstack-core/localstack/services/cloudformation/engine/template_deployer.py]: After some triage I found that the ref is not necessary, only the PhysicalResourceId is. And the assignment is after the search for the provider returns none. In the code you sent me it appeared that the assigment was inside the `try_load_resource_provider` method but this is not possible due to the function only receiving the resource type string.

**Review by dominikschubert — APPROVED** [review]
LGTM, thanks for addressing the comments. Let's see where this approach will take us. I hope we won't run into to many issues where users might expect certain patterns that the mock reference obviously won't adhere to :+1:

**Pull request merged**